sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java

```diff
@@ -375,7 +375,22 @@ public void run() {
     }
 
     private synchronized void processPendingOperations() {
-        for (LocalDocument localDocument : mLocalDocumentStorage.getPendingOperations(Utils.getUserTableName())) {
+        String table = Utils.getUserTableName();
+        for (LocalDocument localDocument : mLocalDocumentStorage.getPendingOperations(table)) {
+            if (localDocument.isExpired()) {
+                if (localDocument.getOperation().equals(Constants.PENDING_OPERATION_PROCESS_VALUE)) {
```

Comment:
Why are we notifying progress? This will potentially be spammy.

Reply:
it's a special condition :) 
This case happens when user set the writeOptions to no cache:
in this case if the pending operation failed in future, we cannot redo anything, thus we need to notify the operation failed :D and this failed is not like set the writeOptions to INFINITE.

Reply:
I don't understand how it's related to the new constant, and why it's checked only when network comes back.

Reply:
Discussed offline.